src/psyclone/psyGen.py

```diff
@@ -5186,12 +5203,19 @@ class SymbolTable(object):
     Encapsulates the symbol table and provides methods to declare new symbols
     and look up existing symbols. It is implemented as a single scope
     symbol table (nested scopes not supported).
+
+    :param kernel: Reference to the KernelSchedule it belongs.
+    :type kernel: :py:class:`psyclone.psyGen.KernelSchedule` or NoneType
     '''
-    def __init__(self):
+    # TODO: (Issue #321) Explore how the SymbolTable overlaps with the
+    # NameSpace class functionality.
+    def __init__(self, kernel=None):
         # Dict of Symbol objects with the symbol names as keys.
         self._symbols = {}
         # Ordered list of the arguments.
         self._argument_list = []
+        # Reference to KernelSchedule it belongs.
```

Comment:
"... to the KernelSchedule to which this symbol table belongs."

Reply:
done